proposals/flexible-vectors/README.md

```diff
@@ -1,7 +1,1184 @@
+If there is no active lanes, then an empty mask is returned.
+
+- `vec.m8.first(m: vec.m8) -> vec.m8`
+- `vec.m16.first(m: vec.m16) -> vec.m16`
+- `vec.m32.first(m: vec.m32) -> vec.m32`
+- `vec.m64.first(m: vec.m64) -> vec.m64`
+- `vec.m128.first(m: vec.m128) -> vec.m128`
+
+```python
+def vec.S.first(m):
+    result = mask.S.New()
+    for i in range(vec.S.length):
+        if m[i]:
+            result[i] = 1
+            break
+    return result
+```
+
+### Mask last
+
+Returns a mask with a single active lane that corresponds to the last active lane of the input.
+If there is no active lanes, then an empty mask is returned.
+
+- `vec.m8.last(m: vec.m8) -> vec.m8`
+- `vec.m16.last(m: vec.m16) -> vec.m16`
+- `vec.m32.last(m: vec.m32) -> vec.m32`
+- `vec.m64.last(m: vec.m64) -> vec.m64`
+- `vec.m128.last(m: vec.m128) -> vec.m128`
+
+```python
+def vec.S.last(m):
+    result = mask.S.New()
+    idx = -1
+    for i in range(vec.S.length):
+        if m[i]:
+            idx = i
+    if idx >= 0:
+        result[idx] = 1
+    return result
+```
+
+
+## Memory operations
+
+### Vector load
+
+- `vec.v8.load(a: memarg) -> vec.v8`
+- `vec.v16.load(a: memarg) -> vec.v16`
+- `vec.v32.load(a: memarg) -> vec.v32`
+- `vec.v64.load(a: memarg) -> vec.v64`
+- `vec.v128.load(a: memarg) -> vec.v128`
+
+### Vector load mask zero
+
+Inactive elements are set to `0`
+
+- `vec.v8.load_mz(m: vec.m8, a: memarg) -> vec.v8`
+- `vec.v16.load_mz(m: vec.m16, a: memarg) -> vec.v16`
+- `vec.v32.load_mz(m: vec.m32, a: memarg) -> vec.v32`
+- `vec.v64.load_mz(m: vec.m64, a: memarg) -> vec.v64`
+- `vec.v128.load_mz(m: vec.m128, a: memarg) -> vec.v128`
+
+### Vector load mask undefined
+
+Inactive elements have undefined values
+
+- `vec.v8.load_mx(m: vec.m8, a: memarg) -> vec.v8`
+- `vec.v16.load_mx(m: vec.m16, a: memarg) -> vec.v16`
+- `vec.v32.load_mx(m: vec.m32, a: memarg) -> vec.v32`
+- `vec.v64.load_mx(m: vec.m64, a: memarg) -> vec.v64`
+- `vec.v128.load_mx(m: vec.m128, a: memarg) -> vec.v128`
+
+### Vector load splat
+
+- `vec.v8.load_splat(a: memarg) -> vec.v8`
+- `vec.v16.load_splat(a: memarg) -> vec.v16`
+- `vec.v32.load_splat(a: memarg) -> vec.v32`
+- `vec.v64.load_splat(a: memarg) -> vec.v64`
+- `vec.v128.load_splat(a: memarg) -> vec.v128`
+
+### Vector store
+
+- `vec.v8.store(a: memarg, v: vec.v8)`
+- `vec.v16.store(a: memarg, v: vec.v16)`
+- `vec.v32.store(a: memarg, v: vec.v32)`
+- `vec.v64.store(a: memarg, v: vec.v64)`
+- `vec.v128.store(a: memarg, v: vec.v128)`
+
+### Vector store mask
+
+Inactive elements are not stored
+
+- `vec.v8.m_store(m: vec.m8, a: memarg, v: vec.v8)`
+- `vec.v16.m_store(m: vec.m16, a: memarg, v: vec.v16)`
+- `vec.v32.m_store(m: vec.m32, a: memarg, v: vec.v32)`
+- `vec.v64.m_store(m: vec.m64, a: memarg, v: vec.v64)`
+- `vec.v128.m_store(m: vec.m128, a: memarg, v: vec.v128)`
+
+## Lane operations
+
+### Splat scalar
+
+`idx` is interpreted modulo the length of the vector.
+
+- `vec.i8.splat(v: vec.v8, x: i32) -> vec.v8`
```

Comment:
What's the purpose of the first parameter? Shouldn't it just be removed?

Reply:
Copy-paste issue, now fixed. Thanks.